src/main/features/core/tray.js

```diff
@@ -39,11 +39,66 @@ const setContextMenu = () => {
 };
 setContextMenu();
 
+
+/**
+ * Toggle the main window (on tray click).
+ * If in background, bring to foreground.
+ */
+function toggleMainWindow() {
+  // the mainWindow variable will be GC'd
+  // we must find the window ourselves
+  const win = WindowManager.getAll('main')[0];
+
+  if (process.platform === 'darwin') { // OS-X - Not tested!
+    if (!win.isVisible()) {
+      // Show
+      win.setSkipTaskbar(false);
+      win.show();
+    } else {
+      // Hide to tray, if configured
+      if (Settings.get('minToTray', true)) {
+        win.hide();
+      }
+    }
+  } else { // Windows, Linux
+    const wasMinimized = win.isMinimized();
+
+    if (wasMinimized || !win.isFocused()) {
+      win.setSkipTaskbar(false);
+      win.show();
+
+      win.focus(); // if not minimized, try to focus it
```

Comment:
I've removed the whole focus thing, because it didn't work due to the other bug (can't get focus after closing the window with a cross button).